FerramAerospaceResearch.Base/Utils/Metrics.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+
+namespace FerramAerospaceResearch.Utils
+{
+    public class Metrics
+    {
+        public Dictionary<string, MetricsElement> data = new Dictionary<string, MetricsElement>();
+        const int hysteresisFactor = 20;
+        public Metrics() { }
+        public void Reset() => data.Clear();
+        public void AddMeasurement(string name, double t)
+        {
+            if (!data.ContainsKey(name))
+            {
+                data.Add(name, new MetricsElement());
+            }
+            MetricsElement m = data[name];
```

Comment:
Double lookup, use `TryGet` and create a new entry if not found

Reply:
Fixed in 8d7866e